src/main/java/uk/gov/hmcts/darts/audio/component/impl/AddAudioRequestMapperImpl.java

```diff
@@ -6,27 +6,25 @@
 import uk.gov.hmcts.darts.audio.model.AddAudioMetadataRequest;
 import uk.gov.hmcts.darts.common.entity.CourtroomEntity;
 import uk.gov.hmcts.darts.common.entity.MediaEntity;
-import uk.gov.hmcts.darts.common.repository.CourtroomRepository;
-
-import java.util.Optional;
+import uk.gov.hmcts.darts.common.service.RetrieveCoreObjectService;
 
 @RequiredArgsConstructor
 @Component
 public class AddAudioRequestMapperImpl implements AddAudioRequestMapper {
 
-    private final CourtroomRepository courtroomRepository;
+    private final RetrieveCoreObjectService retrieveCoreObjectService;
 
     public MediaEntity mapToMedia(AddAudioMetadataRequest addAudioMetadataRequest) {
```

Comment:
Pretty sure your top level service needs `@Transactional` on it if you are wanting to save multiple entities (Media and Courtroom) in the same transaction now. Please have a look at `AudioServiceImpl#addAudio`

Reply:
ACCEPT